core/services/ocr2/plugins/llo/integration_test.go

```diff
@@ -2112,3 +2104,18 @@ func pad32bytes(d uint32) [32]byte {
 	binary.BigEndian.PutUint32(result[28:], d)
 	return result
 }
+
+func newSingleABIEncoder(typ string, multiplier *ubig.Big) (enc lloevm.ABIEncoder) {
+	if multiplier == nil {
+		err := json.Unmarshal([]byte(fmt.Sprintf(`{"type":"%s"}`, typ)), &enc)
+		if err != nil {
+			panic(err)
+		}
+		return
+	}
+	err := json.Unmarshal([]byte(fmt.Sprintf(`{"type":"%s","multiplier":"%s"}`, typ, multiplier.String())), &enc)
+	if err != nil {
+		panic(err)
+	}
+	return
+}
```

Comment:
Why is this implementation so much more complicated than the other one by the same name? Would it be simpler to export a constructor instead?

Reply:
I thought about it, but you never actually need an explicit constructor for this thing outside of tests. You always unmarshal it from JSON.

This one is more complicated because its a blackbox test so we don't have access to the internals of ABIEncoder. And, indeed, from an outside perspective the only way to instantiate one should be via json unmarshalling.